interim-jan-2021/resolver-behavior-self-description.md

```diff
@@ -0,0 +1,27 @@
+# Resolver behavior self-description
+
+## Defining local-only namespaces
+* Express namespaces which only this resolver can resolve
+* Ex 1: Enterprise resolvers serving corporation-specific namespaces
+* Ex 2: Public Wi-Fi or cellular networks offering network-local services
+
+## Defining per-namespace optimization
+* Express namespaces for which this resolver provides preferable resolutions
+* Ex 1: ISP routes public content requests to network caches
+* Ex 2: Cellular provider charges less for content from specific servers
```

Comment:
Is this a thing?  I've never heard of DNS being used this way.

Reply:
One concrete example of this is Netflix Open Connect, which allows ISPs to acquire hardware from Netflix to be hosted in their local data centers and then control routing from their customers so they can serve Netflix content from their own data centers.

Reply:
Sure, but I've never heard of a cellular provider that charges their customers less for content loaded from the local Netflix cache node than they do for content from the Netflix backend servers.

Reply:
Ah, I misread which line was being commented on and was responding to Ex 1. Ignore my previous comment.

I was describing cases where name-based policies are used to determine network interfaces for cellular devices so that some requests work regardless of billing plan state (messages, billing portal, etc.). Obviously this can be solved device-side as well. 

It's come up in vague details in prior EDDI/ADD discussions, but I can't defend it more specifically. Happy to remove and let the WG argue it in from actual cellular folks.

Reply:
It sounds like you're describing the case of a network (cellular plan that ran out of credit) that has access to a very limited subset of the internet.  I think in this case, the limitation is a property of the network, not the resolver, and asking a resolver about a network's behavior is out of scope.

In general, this sounds like a better fit for CAPPORT.

Reply:
> I think in this case, the limitation is a property of the network, not the resolver

I agree. I'll remove this line.

Reply:
I've issued a new commit with the discussed changes. 

Having said that, I just noticed CAPPORT has been concluded. This and other similar scenarios ADD is bringing to the surface (*cough* networks expressing DNS policy *cough*) will need a home that apparently does not exist.

Reply:
An apt example could be: Cellular provider provides regional specific content (for example TV shows accessible to only users of a specific geographic location).